packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+          asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: 'radix' },
+        },
+      ],
+      to: [
+        {
+          to: depositAccount,
+          amount: baseAmount(withdrawAmount),
+          asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: 'radix' },
+        },
+      ],
+      date: confirmed_at,
+      type: TxType.Transfer,
+      hash: intent_hash,
+      asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: 'radix' },
+    }
+    return transaction
+  }
+
+  /**
+   * Creates a transaction using the SimpleTransactionBuilder, signs it with the
+   * private key and returns the signed hex
+   * @param params - The transactions params
+   * @returns A signed transaction hex
+   */
+  async transfer(params: TxParams): Promise<string> {
+    const networkId = this.getRadixNetwork()
+    const radixPrivateKey = this.getRadixPrivateKey()
+
+    if (params.asset == undefined) {
+      throw new Error('asset can not be undefined')
+    }
+    const preparedTransaction = await this.prepareTx(params)
+    const manifest: TransactionManifest = {
+      instructions: { kind: 'String', value: preparedTransaction.rawUnsignedTx },
+      blobs: [],
+    }
+    try {
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const notarizedTransaction = await TransactionBuilder.new().then((builder) =>
+        builder
+          .header({
+            networkId: networkId,
+            startEpochInclusive: gatewayStatusResponse.ledger_state.epoch,
+            endEpochExclusive: gatewayStatusResponse.ledger_state.epoch + 10,
+            nonce: generateRandomNonce(),
+            notaryPublicKey: radixPrivateKey.publicKey(),
+            notaryIsSignatory: true,
+            tipPercentage: 0,
+          })
+          .manifest(manifest)
+          .notarize(radixPrivateKey),
+      )
+
+      const compiledTransaction = await RadixEngineToolkit.NotarizedTransaction.compile(notarizedTransaction)
+      const compiledTransactionHex = Convert.Uint8Array.toHexString(compiledTransaction)
+      await this.broadcastTx(compiledTransactionHex)
+      const transactionId = await RadixEngineToolkit.NotarizedTransaction.intentHash(notarizedTransaction)
+      return transactionId.id
+    } catch (error) {
+      throw new Error('Failed to transfer')
+    }
+  }
+  /**
+   * Submits a transaction
+   * @param txHex - The transaction hex build with the transfer method
+   * @returns - The response from the gateway
+   */
+  async broadcastTx(txHex: string): Promise<string> {
+    const transactionSubmitResponse: TransactionSubmitResponse =
+      await this.gatewayApiClient.transaction.innerClient.transactionSubmit({
+        transactionSubmitRequest: {
+          notarized_transaction_hex: txHex,
+        },
+      })
+    return JSON.stringify(transactionSubmitResponse)
```

Comment:
This does not conform to the interface defined by XChainJs. The interface is:

```ts
abstract broadcastTx(txHex: string): Promise<TxHash>
```

This means that the return should be a transaction hash but the current impl returns a gateway response which is incorrect. So, we should modify this function to return a tx hash.

Reply:
done